docs/csharp/how-to/compare-strings.md

```diff
@@ -0,0 +1,86 @@
+---
+title: "How to: Compare strings - C# Guide"
+description: Learn how to compare and order string values, with or without case, with or without culture specific ordering
+ms.date: 03/13/2018
+ms.prod: .net
+ms.technology: 
+  - "devlang-csharp"
+ms.topic: "article"
+helpviewer_keywords: 
+  - "strings [C#], comparison"
+  - "comparing strings [C#]"
+author: "BillWagner"
+ms.author: "wiwagn"
+---
+# How to compare strings in C# #
+
+When you compare strings, you define an order among them. Comparisons are used to sort a sequence of strings. Once the sequence is in a known order, it is easier to search, both for software and for humans. Other comparisons may check if strings are the same. These sameness checks are similar to equality, but some differences, such as case differences may be ignored.
+
+The methods that compare strings indicate that one string is greater than or less than the other, or that the two strings are equal. The comparison result depends on rules that specify performing an *ordinal comparison* or a *culture-sensitive comparison* and if case should be considered. You pick the rules that match your scenario.
+
+Use basic ordinal comparisons when you have to compare two strings without regard to linguistic conventions. A basic ordinal comparison (<xref:System.StringComparison.Ordinal?displayProperty=nameWithType>) is case-sensitive, which means that the two strings must match character for character: "and" does not equal "And" or "AND". A frequently used variation is <xref:System.StringComparison.OrdinalIgnoreCase?displayProperty=nameWithType>, which will consider "and", "And", and "AND" all equal. <xref:System.StringComparison.OrdinalIgnoreCase> is often used to compare file names, path names, network paths, and any other string whose value does not change based on the locale of the user's computer.
```

Comment:
One can read the term *compare* in two ways (@BillWagner  you already mentioned that in the first paragraph):
- figure out ordering of two strings
- figure out if two strings are equal or not

If a paragraph talks mostly about one meaning, I suggest to specify that.
Here, it looks, we talk only about *equal/not equal* meaning. As for sorting, I'm surprised to find that ordering changes:
```csharp
Console.WriteLine(string.Compare("and", "And", StringComparison.Ordinal));           // gives 32
Console.WriteLine(string.Compare("and", "And", StringComparison.InvariantCulture));  // gives -1
```